### PR TITLE
Test Service creation -> endpoint generation latency

### PR DESCRIPTION
**Pull request by lavalamp**
To measure the problems reported in #9122.

This test passes. Not sure if the latency pass/fail condition is overly strict or not.

### CODE REVIEW

File: test/e2e/service_latency.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (rjnagal) [inline test/e2e/service_latency.go]: 2015

File: test/e2e/service_latency.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+		}
+		if e, ok := val.Object.(*api.Endpoints); ok {
+			if e.Name == cfg.Name && len(e.Subsets) > 0 && len(e.Subsets[0].Addresses) > 0 {
+				stopTime := time.Now()
+				return stopTime.Sub(startTime), nil
+			}
+		}
+	}
+
+	// Now make a service and see how long it takes to have its endpoints show up.
```
Comment (rjnagal) [inline test/e2e/service_latency.go]: misplaced comment.

**Comment by rjnagal** [comment]
I think the initial latency limits look fine. I'd expect it to pass every time :)

just small nits, otherwise LGTM.

**Comment by lavalamp** [comment]
Comments addressed, thanks

**Comment by rjnagal** [comment]
LGTM

**Comment by krousey** [comment]
@lavalamp This isn't merged yet because #9122 is still v1-candidate. What should be done here? Should we mark this as post-v1 until it's decided? Or just let it sit in the merge queue for now?

**Comment by lavalamp** [comment]
Additional testing is OK for v1.

**Comment by brendandburns** [comment]
Massive flakiness in e2e, looking like it's due to this PR.  Reverting.

**Comment by brendandburns** [comment]
Details from the test log:
```
•! Panic [0.320 seconds]
Pods
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:679
  should be submitted and removed [It]
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pods.go:256

  Test Panicked
  runtime error: invalid memory address or nil pointer dereference
  /usr/src/go/src/runtime/asm_amd64.s:401

  Full Stack Trace
  /usr/src/go/src/runtime/panic.go:42 (0x41452e)
  /usr/src/go/src/runtime/sigpanic_unix.go:26 (0x41ace4)
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service_latency.go:129 (0x4902f6)
  /go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service_latency.go:102 (0x4b7d2f)
  /usr/src/go/src/runtime/asm_amd64.s:2232 (0x440671)
```

**Pull request merged**